src/link_manager/LinkManagerStateMachineActiveActive.cpp

```diff
@@ -722,14 +723,16 @@ void ActiveActiveStateMachine::switchMuxState(
 //
 void ActiveActiveStateMachine::switchPeerMuxState(mux_state::MuxState::Label label)
 {
-    MUXLOGWARNING(
-        boost::format("%s: Switching peer MUX state to '%s'") %
-        mMuxPortConfig.getPortName() %
-        mMuxStateName[label]
-    );
-    enterPeerMuxState(label);
-    mMuxPortPtr->setPeerMuxState(label);
-    startPeerMuxWaitTimer();
+    if (mMuxPortConfig.getMode() == common::MuxPortConfig::Mode::Auto) {
+        MUXLOGWARNING(
+            boost::format("%s: Switching peer MUX state to '%s'") %
+            mMuxPortConfig.getPortName() %
+            mMuxStateName[label]
+        );
+        enterPeerMuxState(label);
+        mMuxPortPtr->setPeerMuxState(label);
+        startPeerMuxWaitTimer();
```

Comment:
for the detach mode, could you please point , where is the actual logic to not query the peer ToR's gRPC API's ?